src/engraving/playback/playbackmodel.cpp

```diff
@@ -369,21 +369,51 @@ void PlaybackModel::processSegment(const int tickPositionOffset, const Segment*
             continue;
         }
 
-        if (item->type() == ElementType::MEASURE_REPEAT) {
-            const MeasureRepeat* measureRepeat = toMeasureRepeat(item);
-            const Measure* currentMeasure = measureRepeat->measure();
-            const Measure* referringMeasure = measureRepeat->referringMeasure();
+        if (isFirstSegmentOfMeasure) {
+            auto processMeasureRepeat
```

Comment:
You should probably move this code into a separate private method instead of using the lambda